### PR TITLE
eex engine raises on missing assigns

### LINKED ISSUE #3327: EEx should raise on missing assigns
We should follow up on what Phoenix decides:

https://github.com/phoenixframework/phoenix/issues/879

### PR DESCRIPTION
**Pull request by tony612**
resolve #3327 
The code is almost the same with that in phoenix_html except I changed `fetch_assign` to `fetch_assign!`

### CODE REVIEW

**Comment by whatyouhide** [comment]
I'd go with "EEx" instead of "sex" but that's a personal preference :)

**Comment by tony612** [comment]
@whatyouhide Sorry about the typo(it seems my gitx correct it automatically) :joy: It's OK now XD

**Comment by whatyouhide** [comment]
Don't worry, I was kidding :) :laughing:

**Comment by tony612** [comment]
@josevalim How about this PR? Just realized that it's still open :smile:

File: lib/eex/lib/eex/engine.ex [hunk 1]
```diff
@@ -66,13 +66,23 @@ defmodule EEx.Engine do
...
     arg
   end
 
+  @doc false
+  def fetch_assign!(assigns, key) do
+    case Dict.fetch(assigns, key) do
+      :error ->
+        raise ArgumentError, message: """
+        assign @#{key} not available in eex template. Available assigns: #{inspect Dict.keys(assigns)}
+        """
```
Comment (ericmj) [inline lib/eex/lib/eex/engine.ex]: This will create a newline after the error message, use a normal string instead.

**Comment by ericmj** [comment]
Just a single comment, rest looks good to me.

**Comment by tony612** [comment]
@ericmj Done and pushed with force :smile:

**Pull request merged**